btrfsmaintenance-functions

```diff
@@ -66,3 +66,29 @@ check_balance_running() {
 	fi
 	return 0
 }
+
+# function: wait_ac_power
+# parameter: {timeout} {AC power online device}
+#
+# wait until ac power goes online
+wait_ac_power() {
+	local timecount=0
+	local timeout=0
+	local ac_power_device=/sys/class/power_supply/AC/online
```

Comment:
How reliable is this? The file does not exist on my system. Though there's a user-defined path for that I don't think we should require to go to /sys and let the user find the right one. I don't see any "stable ABI" path defined in the kernel docs so we'd need at least some heuristic for the path of fallback.

Reply:
yep, I'm still thinking about a way to grab the information without use external tools (like acpi client) and reading from /sys directly.

Reply:
https://github.com/kdave/btrfsmaintenance/pull/45#issuecomment-363898558

That said, I seem to remember reading that the /sys interface for power was somewhat depreciated...IIRC acpi client or upower were the recommended alternatives.  'could be wrong about that, as the memory is very vague!